lib/dateTime.js

```diff
@@ -216,16 +308,82 @@ function formatDateString(date) {
 function getDateStringWithTimezone(date, timezone) {
 	let dateString = `${date.month}/${date.date}/${date.year}, ${date.hours}:${date.minutes}:${date.seconds} ${timezone}`;
 	if (timezone.includes('-') || timezone.includes('+')) {
-		const re = /([A-Z]{3})(\+|-)([0-9]{1,2}):([0-9]{2})/;
+		const re = /([A-Z]{3})?(\+|-)([0-9]{1,2})(:)?([0-9]{2})?/;
 		const match = timezone.match(re);
-		if (match && match.length === 5) {
+		if (match && match.length === 6) {
 			// YYYY-MM-DDTHH:mm:ss(+|-)HH:MM
-			dateString = `${date.year}-${date.month}-${date.date}T${date.hours}:${date.minutes}:${date.seconds}${match[2]}${prePadByZero(match[3], 2)}:${prePadByZero(match[4], 2)}`;
+			const mins = match[5] ? prePadByZero(match[5], 2) : '00';
+			dateString = `${date.year}-${date.month}-${date.date}T${date.hours}:${date.minutes}:${date.seconds}${match[2]}${prePadByZero(match[3], 2)}:${mins}`;
 		}
 	}
 	return dateString;
 }
 
+function getTimezoneFromDuplicatedAbbreviation(abbrTimezone) {
+	const longTimezone = getDocumentLocaleSettings().timezone.identifier;
+	switch (abbrTimezone) {
+		case 'ADT':
+			// Arabia Daylight Time +0400 (Asia)
+			// Atlantic Daylight Time -0300 (America)
```

Comment:
In the web site you listed, it shows other abbreviations for Halifax -- ADST, AST and HAA -- I'm assuming those don't get used by the browsers as they're not in the map?

Reply:
I think those aren't used very frequently, though I can add them just in case. I did start adding the alternatives but it must have been later on in the file and I missed going back to the top.